providers/os/connection/docker_container.go

```diff
@@ -225,11 +225,15 @@ func NewContainerRegistryImage(id uint32, conf *inventory.Config, asset *invento
 		conn.PlatformIdentifier = identifier
 		conn.Metadata.Name = containerid.ShortContainerImageID(hash.String())
 
-		asset.PlatformIds = []string{identifier}
 		repoName := ref.Context().Name()
 		imgDigest := hash.String()
 		name := repoName + "@" + containerid.ShortContainerImageID(imgDigest)
-		asset.Name = name
+		if asset.Name == "" {
+			asset.Name = name
+		}
+		if len(asset.PlatformIds) == 0 {
```

Comment:
Why do we need that check?

Reply:
In the case I was testing, the platformID was already set to the correct sha, the same as for the asset name.

As I'm not sure whether we end up here with an empty name or platformID, I added the `if`s instead of removing this code part.

Reply:
1. It is a different platform id, then we need to figure out why, but this can happen if users define their own
2. We should only add the platform id that we determined, when its different to the list so that we ensure uniqness (i think we validate this later, but we need to double-check)
3. We should definitely always add the platform id as long as it is different, not just when `len(asset.PlatformIds) == 0` is matched

Would love to see a test case for that since this seems to be an error case we can run more often in

Reply:
As discussed, I now append the platformID in case there is already one present.

I also extended the test case to check for the platformID, I'd expect for this remote image.